### ISSUE TITLE
[bug] Memory issue

### ISSUE THREAD

**Issue by nailyk-fr** [tootsuite/mastodon#6438]
Hi, 

in case you scroll down a bit the 'first' column (timeline? fediverse in french), and do not touch it anymore, all new toot will be loaded on top of this, keeping the scroll bar growing. 
![capture d ecran_2018-02-07_11-37-10](https://user-images.githubusercontent.com/16372405/35912151-058118ba-0bfc-11e8-8a97-691c2d7247d8.png)

Not a big issue but this behavior will cause a lot of memory drain, while all newer toots gets loaded. 

![before](https://user-images.githubusercontent.com/16372405/35912143-030d112e-0bfc-11e8-9007-9b3904ab2c4c.png)

* * * *

- [x] I searched or browsed the repo’s other issues to ensure this is not a duplicate.
- [x] This bug happens on a [tagged release](https://github.com/tootsuite/mastodon/releases) and not on `master` (If you're a user, don't worry about this).

**Comment by vdeflorio** [comment]
Exactly! If I scroll the Home window a little, then the system starts accumulating notifications. This is normal -- the intended behavior. Still, the system slows down tremendously -- it becomes more and more irresponsive.

**Comment by defaultxr** [comment]
I'm also being affected by this; apparently I forget to scroll back up to the top a lot because this issue happens to my Mastodon tab pretty often and it's annoying as you might imagine.

This issue doesn't happen for Tweetdeck, which has a similar UI. It seems like after a while Tweetdeck will collapse/discard posts in a column between the current scroll position and the top. Might be a good solution for this issue, since most of the time if I'm far enough from the top of a column, I usually just click the column header to warp back to the top.

Tweetdeck also makes it possible to continue from the current position; if you scroll up and reach the point the posts were removed from, there is a button to "load more tweets" which loads the skipped ones and puts them back in the timeline so you can continue reading from that point.

Also, in my experience this affects any column, not just the first one.

**Comment by Dormouse759** [comment]
This has become quite a big issue with the Fediverse growing due to recent events.

When in the federated tab, Mastodon keeps loading up a lot of new posts and can easilly crash the tab, running out of memory.

**Comment by Dormouse759** [comment]
This seems like the very same bug, just taken from a different point of view: #19991

**Comment by afontenot** [comment]
> This seems like the very same bug, just taken from a different point of view: #19991

I think it's best to keep this bug about the memory issues resulting from automatically loading new posts indefinitely, and the other bug is about a specific visual glitch that results from loading new posts. Maybe they will be fixed in the same way, but maybe not.

**Comment by prplecake** [comment]
When you're scrolled to the top, as new items come in it seems old items "fall off" the other end, keeping the total number of items at a sane number. When you're scrolled down at all, enough so you're not at the "top" of the feed, old items don't get removed from the list.

Scrolling around other timelines doesn't seem particularly bothersome, but if you try to ~~send a toot~~ publish a post, you'll probably hit #22225. Refreshing the page works to get things back into working order, although it takes some time for this refresh to actually complete since the tab is basically locked up.

**Comment by mjankowski** [comment]
Is this still relevant (and as noticeable) on 4.3 release and with modern browsers? Anyone have updated metrics?

**Comment by prplecake** [comment]
Memory still steadily grows, but the performance impact seems lesser.

**Comment by heyitsbrib** [comment]
Is this still noticeable in 2026?

**Comment by wargreen** [comment]
> Is this still noticeable in 2026?
Yes, i have 2 instances in pinned tab (firefox latest, Debian SID) in my 2 computers (so 4 tab in total)

I can confirm that i need to kill each tab each week because memory issues.
Sometime 8 or 10 Go each tab. On the 16Go laptop and 32Go desktop, it is really too much.
So i think that this issue need a strongest priority.

We need a kind of grabage colector. Maybe, for each toot, store the time of "last viewed", and unload it if the tab consum a certain percentage of the system memory ?

**Comment by wargreen** [comment]
If that can be relevant, here a part of my firefox's `about:memory` details of the mastodon tab : 

<img width="787" height="218" alt="Image" src="https://github.com/user-attachments/assets/27d947f3-641a-4075-a199-e0129e7b4850" />

`about:processes` show 3Go of memory usage. After unload then reload the tab, 157Mo.
